src/main/resources/public/ts/utilities/availability.util.ts

```diff
@@ -130,22 +131,31 @@ export class AvailabilityUtil {
     };
 
     /**
-     * Calculate quantity used by the bookings overlapping our given timeslot
+     * Calculate max quantity used by the bookings overlapping our given timeslot
      *
      * @param booking       booking giving timeslot to check
      * @param resource      resource of the booking
      * @param exception     booking not to count for calculus
      * @return number       quantity used by the bookings using this timeslot
      */
-    static getBookingsQuantityUsedOnTimeslot = (booking: any, resource: any, exception?: any) : number => {
-        let bookings = resource.bookings.all;
-        let bookingsQuantityUsed = 0;
-        let exceptionId = (exception && exception.id) ? exception.id : -1;
-
-        for (let b of bookings) {
-            if (b.id != exceptionId && b.parent_booking_id != exceptionId &&
-                b.status != 3 && !b.is_periodic && BookingUtil.isBookingsOverlapping(booking, b)) {
-                    bookingsQuantityUsed += b.quantity;
+    static getBookingsQuantityUsedOnTimeslot = (booking: Booking, resource: Resource, exception?: Booking) : number => {
+        let siblings: Booking[] = BookingUtil.getSiblings(booking, resource);
+        let bookingsQuantityUsed: number = 0;
+        let exceptionId: number = (exception && exception.id) ? exception.id : -1;
+
+        for (let sibling of siblings) {
+            if (sibling.id != exceptionId && sibling.parent_booking_id != exceptionId && sibling.status != 3) {
+                let localSiblings: Booking[] = BookingUtil.getSiblings(sibling, resource);
+                let localMaxQuantityUsed: number = 0;
+                for (let localSibling of localSiblings) {
+                    if (localSibling.id != exceptionId && localSibling.id != booking.id
+                        && localSibling.parent_booking_id != exceptionId && localSibling.status != 3
+                        && BookingUtil.isBookingsOverlapping(localSibling, booking)) {
+                            localMaxQuantityUsed += localSibling.quantity;
+                    }
+                }
+                localMaxQuantityUsed += sibling.quantity;
+                if (localMaxQuantityUsed > bookingsQuantityUsed) bookingsQuantityUsed = localMaxQuantityUsed;
             }
```

Comment:
```suggestion
        let siblingsLocalMaxQuantityUsed : Array<number> = siblings
            .filter(sibling => sibling.id != exceptionId && sibling.parent_booking_id != exceptionId && sibling.status != 3)
            .map(sibling => {
                let localSiblings: Booking[] = BookingUtil.getSiblings(sibling, resource);
                let localMaxQuantityUsed: number = 0;
                localSiblings
                    .filter(localSibling => localSibling.id != exceptionId && localSibling.id != booking.id
                        && localSibling.parent_booking_id != exceptionId && localSibling.status != 3
                        && BookingUtil.isBookingsOverlapping(localSibling, booking))
                    .forEach(localSibling => localMaxQuantityUsed += localSibling.quantity);
                localMaxQuantityUsed += sibling.quantity;
                return localMaxQuantityUsed;
            })

        bookingsQuantityUsed = Math.max(...siblingsLocalMaxQuantityUsed);
```
Une version sans les boucle. A voir si c'est pour autant mieux